sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/regexpExpressions.scala

```diff
@@ -543,25 +543,52 @@ case class RLike(left: Expression, right: Expression) extends StringRegexExpress
 case class StringSplit(str: Expression, regex: Expression, limit: Expression)
   extends TernaryExpression with ImplicitCastInputTypes with NullIntolerant {
 
-  override def dataType: DataType = ArrayType(StringType, containsNull = false)
-  override def inputTypes: Seq[DataType] = Seq(StringType, StringType, IntegerType)
+  override def dataType: DataType = ArrayType(str.dataType, containsNull = false)
+  override def inputTypes: Seq[AbstractDataType] =
+    Seq(StringTypeAnyCollation, StringTypeAnyCollation, IntegerType)
+
+  override def checkInputDataTypes(): TypeCheckResult = str.dataType match {
+    case s: StringType =>
+      if (s.supportsBinaryEquality || s.isUTF8BinaryLcaseCollation) {
```

Comment:
we shouldn't do this manually for every function
instead, a more general way to handle this has been introduced in `CollationTypeConstraints.scala`

use the appropriate case object stemming from `StringTypeCollated`
(in this case: StringTypeBinary)